Plugins/AWSLambdaDeployer/Plugin.swift

```diff
@@ -0,0 +1,348 @@
+                arguments: configuration.products.compactMap { $0.name },
+                logLevel: configuration.verboseLogging ? .debug : .silent)
+            try samDeploymentDescriptor.write(
+                to: samDeploymentDescriptorUrl, atomically: true, encoding: .utf8)
+
+            print("-------------------------------------------------------------------------")
+            print("validating SAM deployment descriptor")
+            print("-------------------------------------------------------------------------")
+            try self.execute(
+                executable: samExecutableUrl,
+                arguments: ["validate", "-t", samDeploymentDescriptorUrl.path],
+                logLevel: configuration.verboseLogging ? .debug : .silent)
+
+            if !configuration.noDeploy {
+                print("-------------------------------------------------------------------------")
+                print("deploying AWS Lambda function")
+                print("-------------------------------------------------------------------------")
+                try self.execute(
+                    executable: samExecutableUrl,
+                    arguments: ["deploy", "-t", samDeploymentDescriptorUrl.path],
+                    logLevel: configuration.verboseLogging ? .debug : .silent)
+            }
+        } catch Errors.processFailed(_, _) {
+            print("The generated SAM template is invalid or can not be deployed.")
+            if configuration.verboseLogging {
+                print("File at : \(samDeploymentDescriptorUrl)")
+            } else {
+                print("Run the command again with --verbose argument to receive more details.")
+            }
+        } catch {
+            print("Can not execute file at:")
+            print("\(deploymentDescriptorExecutableUrl.path)")
+            print("or at:")
+            print("\(samExecutableUrl.path)")
+            print("Is SAM installed ? (brew tap aws/tap && brew install aws-sam-cli)")
+            print("Did you add a 'Deploy' executable target into your project's Package.swift ?")
+            print("Did you build the release version ? (swift build -c release)")
+        }
+    }
+
+    @discardableResult
+    private func execute(
+        executable: URL,
+        arguments: [String],
+        customWorkingDirectory: URL? = nil,
+        logLevel: ProcessLogLevel
+    ) throws -> String {
+        try self.execute(
+            executable: Path(executable.path),
+            arguments: arguments,
+            customWorkingDirectory: customWorkingDirectory == nil
+            ? nil : Path(customWorkingDirectory!.path),
+            logLevel: logLevel)
+    }
+
+    // **************************************************************
+    // Below this line, the code is copied from the archiver plugin
+    // **************************************************************
+    @discardableResult
+    private func execute(
```

Comment:
I assume this is copy/paste from the other plugin? maybe we can combine them to a single implementation?

Reply:
Initially I did not want to touch the Archiver plugin.  Now that I know you like the idea of the deploy plugin, I will factor out common codes in a separate shared library

Reply:
I wonder what is the best approach to share code between two Plugins ?  Share a common executable to carry on the actual work ? Symlinks ?

Check  this thread https://forums.swift.org/t/difficulty-sharing-code-between-swift-package-manager-plugins/61690/11

Reply:
yea this is a "hole" in SwiftPM plugins that the team is working to fix afaik, the work around is to put it in a file and symlink it across the targets

Reply:
OK - will do this

Reply:
Done with https://github.com/sebsto/swift-aws-lambda-runtime/commit/8a816c5f34d3aefd4d299b6cde42efa651ad9c02